src/opening-period/page/EditOpeningPeriod.tsx

```diff
@@ -0,0 +1,301 @@
+import { ResourceInfo } from '../../resource/page/ResourcePage';
+import './EditOpeningPeriod.scss';
+
+type FormData = {
+  openingPeriodTitle: string;
+  openingPeriodOptionalDescription: string;
+  openingPeriodBeginDate: string;
+  openingPeriodEndDate: string;
+};
+
+type SubmitStatus = 'init' | 'succeeded' | 'error';
+
+const FormDatePicker = ({
+  name,
+  label,
+  control,
+  register,
+}: {
+  name: string;
+  label: string;
+  control: Control<FormData & FieldValues>;
+  register: () => void;
+}): JSX.Element => (
+  <Controller
+    control={control}
+    name={name}
+    render={({ value, onChange }): JSX.Element => (
+      <Datepicker
+        id={name}
+        dataTest={name}
+        labelText={label}
+        onChange={onChange}
+        value={new Date(value)}
+        registerFn={register}
+        required
+      />
+    )}
+  />
+);
+
+export default function EditOpeningPeriodPage({
+  resourceId,
+  datePeriodId,
+}: {
+  resourceId: string;
+  datePeriodId: string;
+}): JSX.Element {
+  const id = parseInt(datePeriodId, 10);
+  const selectedLanguage: Language = 'fi';
+  const [resource, setResource] = useState<Resource>();
+  const [hasLoadingResourceError, setLoadingResourceError] = useState<
+    Error | undefined
+  >(undefined);
+  const [hasDatePeriodLoadingError, setDatePeriodLoadingError] = useState<
+    Error | undefined
+  >(undefined);
+  const { reset, register, handleSubmit, control, errors } = useForm<
+    FormData
+  >();
+  const history = useHistory();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+  const [submitStatus, setSubmitStatus] = useState<SubmitStatus>('init');
+  const resetForm = useCallback(
+    (datePeriod: DatePeriod): void =>
+      reset({
+        openingPeriodTitle: datePeriod.name[selectedLanguage] || '',
+        openingPeriodOptionalDescription:
+          datePeriod.description[selectedLanguage] || '',
+        openingPeriodBeginDate: datePeriod.start_date || undefined,
+        openingPeriodEndDate: datePeriod.end_date || undefined,
+      }),
+    [reset]
+  );
+
+  const onSubmit = async (data: FormData): Promise<void> => {
+    try {
+      if (!resource || !data) {
+        throw new Error('This should never happen, but typescript');
+      }
+
+      setIsSaving(true);
+      const updatedPeriod = await api.putDatePeriod({
+        resource: resource.id,
+        id,
+        name: {
+          fi: data.openingPeriodTitle,
+          sv: null,
+          en: null,
+        },
+        description: {
+          fi: data.openingPeriodOptionalDescription,
+          sv: null,
+          en: null,
+        },
+        start_date: transformToApiFormat(data.openingPeriodBeginDate),
+        end_date: transformToApiFormat(data.openingPeriodEndDate),
+        resource_state: ResourceState.OPEN,
+        override: false,
+        time_span_groups: [],
+      });
+      resetForm(updatedPeriod);
```

Comment:
Is this resetForm necessary?

Reply:
Maybe the updatedPeriod object could hold some api side validation errors potentially, and we would wish to render those and there resetForm would be required?

Reply:
I think it is a good practice reset form state since API returns the updated value

Reply:
Though it should be the exact same value we put there? hmm